src/server/services/cache/redis.ts

```diff
@@ -0,0 +1,90 @@
+import Redis from 'ioredis'
+
+import { Cache, Key, Value, ValueCallback } from './types'
+import {DEFAULT_CACHE_TTL} from "../../../config";
+
+export const RedisCache = (): Cache => {
+  const redis = new Redis(process.env.REDIS_URL as string)
+    redis.on('connect', () => {
+        console.log('CacheStore - Connection status: connected');
+    });
+    redis.on('end', () => {
+        console.log('CacheStore - Connection status: disconnected');
+    });
+    redis.on('reconnecting', () => {
+        console.log('CacheStore - Connection status: reconnecting');
+    });
+    redis.on('error', (err) => {
+        console.log('CacheStore - Connection status: error ', { err });
+    });
+  return {
+    getAndSet<T extends Value>(
+      key: Key,
+      valueFunction: ValueCallback<T>,
+      ttl: number = DEFAULT_CACHE_TTL,
+    ): Promise<T> {
+      return new Promise<T>((resolve) => {
+        this.get(key).then((value) => {
+          if (value) {
+            resolve(value as T)
+          } else {
+            valueFunction().then((newValue) => {
+              resolve(newValue)
+              this.set(key, newValue, ttl)
+            })
+          }
+        })
+      })
+    },
+    flush(): Promise<void> {
+      return new Promise<void>((resolve, reject) => {
+        redis.flushdb((err) => {
+          if (err) {
+            reject()
+          } else {
+            resolve()
+          }
+        })
+      })
+    },
+    get<T extends Value>(key: Key): Promise<T> {
+      return new Promise((resolve, reject) => {
+        try {
+          redis.hgetall(key, (err, result) => {
+            if (err) {
+              reject()
```

Comment:
Ich würde jeweils direkt auch returnen. Damit der restliche check nicht dennoch ausgeführt wird. Theoretisch ists möglich nen error + results zu erhalten.

Dem reject würde ich den error noch weitergeben, so kann man den catchen und erhält keinen leeren promise reject (Debugging, weitere logik im catch ect)

Reply:
Hmm, also direkt returnen würde ich nicht - aber du hast recht, theoretisch könnten reject und resolve "gleichzeitig" aufgelöst werden. On it.

Reply:
habs versucht zu verbessern

Reply:
Nochmals verbessert